src/main/java/frc/robot/commands/IntakeAlgae.java

```diff
@@ -0,0 +1,21 @@
+package frc.robot.commands;
+
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import frc.robot.subsystems.Gripper;
+
+public class IntakeAlgae extends MoveGenericSubsystem {
+
+    private static final double INTAKE_SPEED = -0.5;
+
+    private final Gripper gripper;
+
+    public IntakeAlgae(Gripper gripper) {
+        super(gripper, INTAKE_SPEED);
+        this.gripper = gripper;
+    }
+
+    @Override
+    public boolean isFinished() {
+        return gripper.hasAlgae();
```

Comment:
`MoveGenericSubsystem` implements an `isFinished` that uses the subsystem's `canMove`.
`canMove` in `MotoredGenericSubsystem` is default to return true, but you can override it how you want (ie. if the speed negative, return false if there is an algae in the system)